spec/controllers/two_factor_authentication/otp_expired_controller_spec.rb

```diff
@@ -61,6 +61,55 @@
     end
 
     context 'otp expired redirect feature is turned off' do
+      it 'does not redirect' do
+        allow(IdentityConfig.store).to receive(:allow_otp_countdown_expired_redirect).
+          and_return(false)
+
+        get :show
+
+        expect(response).to_not render_template(:show)
+      end
+    end
+
+    context 'with a new account' do
+      before do
+        allow(controller).to receive(:new_account_mfa_registration?).and_return(true)
+        allow(controller).to receive(:unconfirmed_phone?).and_return(true)
+      end
+
+      it 'redirects to authentication methods setup' do
+        user = create(:user)
+        stub_sign_in_before_2fa(user)
+
+        get :show
+
+        expect(assigns(:authentication_options_path)).to eq(authentication_methods_setup_url)
+      end
+
+      it 'provides an option to choose another phone number' do
+        user = create(:user)
+        stub_sign_in_before_2fa(user)
+
+        get :show
+        expect(assigns(:use_another_phone_path)).to eq(phone_setup_path)
+      end
+    end
+
+    context 'with an existing account' do
```

Comment:
We may want to break this down to "existing account signing in" and "existing account adding a new phone number".

Reply:
I was thinking about doing this, so I can take care of that.